cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -221,16 +221,19 @@ def is_valid_sample_fastq_filename(self, fastq_file_name: str) -> bool:
         """Validate the file name and discard any undetermined fastq files."""
         return "Undetermined" not in fastq_file_name
 
-    def get_sample_fastq_file_paths(self, flow_cell_directory: Path) -> List[Path]:
-        """Get fastq file paths for flow cell."""
-        valid_sample_fastq_file_paths = [
-            file_path
-            for file_path in flow_cell_directory.glob(
-                f"**/*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
-            )
-            if self.is_valid_sample_fastq_filename(file_path.name)
-        ]
-        return valid_sample_fastq_file_paths
+    def get_flowcell_sample_fastq_file_paths(self, flow_cell_directory: Path) -> List[Path]:
+        """Get fastq file paths for a flow cell."""
+        base_pattern = f"Unaligned*/Project_*/Sample_*/*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
+        alt_pattern = (
+            f"Unaligned*/Project_*/Sample_*_*/*{FileExtensions.FASTQ}{FileExtensions.GZIP}"
+        )
```

Comment:
Will not all matches in alt_pattern match base_pattern as well? I would guess that x*_*y would match with x*y since * is a wildcard?

Reply:
Nice catch! I think you are right